### PR TITLE
[TESTMERGE WAS A DISASTER]Bobmed V1: A currently partial port of paradise medicine to fit TG standards.

### PR DESCRIPTION
**Pull request by BobJoga**
## About The Pull Request

Title.
Current features:
- Nerfed cloning. Cloning requires biomass to be performed, which takes 6 slabs of normal meat or 3 slabs of biomeat, which can be produced in chemistry.
- Limb separation. Hands and feet aren't just part of the arm of leg, they're actually separate bodyparts - same for the groin.
- Broken bones. Limbs can be broken with enough luck and damage. Broken bones can only truly be fixed with surgery, and leave that limb completely disabled until the broken bone is fixed. Splints serve as a temporary solution, however. This only applies to organic limbs.
- Internal bleeding. With enough luck or damage, limbs can have internal bleeding. This can also only be truly fixed with surgery, and has no temporary solution - not even gauze is gonna work. This only applies for organic limbs.
- Internal organ damage. If an attack is strong enough, it will not only damage the limbs, but also the organs inside it.
- Synthetic limbs cannot be broken or have internal bleeding - but are more frail and disable with less damage. They can be healed both with the shit in Azarak's update, and nanopaste.
- Surgeries for the new ailments above.

## Why It's Good For The Game

Medicine too easy. Fights have no lasting effects, everything can be fixed with a medkit.

## Changelog
:cl:
add: Medicine overhaul. See PR #1701 for more details.
/:cl:

## Well shit it got testmerged - Bobmed V1 Guide

The additions have been detailed above. This will just cover how to fix them and things you might need.

### Unsorted problem areas
**Mining** - For the moment there is no bone healing juice reagent, so your best bet is just straight up not messing with megafauna (or just normal fauna, even) when you have no spare cores. I know that being a wussy isn't the miner way, but trust me you don't want a goliath to break your legs and make you crawl back to the base. As implied, legion cores will fix internal bleeding and broken limbs immediately.
**Cloning** - Being a clooner is no longer that viable. Cloning will take biomass, which you gather by throwing meat slabs on the cloner and letting it suck it up. There is a caveat, however - Most types of meat are inefficient, and will take 6 slabs to clone a single person. This is why you should instead create biomeat, which only takes 3 slabs to clone someone *(5 blood, 5 cryoxadone, 20 synthflesh, 10 nutriment [catalyst] )*. Cloner biomass requirements and gains are fixed and not affected by better parts, which is intentional, but not immutable. I might change it in the future if i deem it necessary.
**Medical in general** - The sleeper and health analyzer are now way more important as tools of the trade. Sleepers will not only heal minor injuries with chems, as they did before, but they also serve as an analysis tool. With them, you can know which limbs are missing, which organs are missing, and which organs or limbs are damaged (including specific information such as internal bleeding and fractures).
**Surgery in general** - Surgeries which require any kind of incision will leave the patient bleeding, for obvious reasons. This is why you cannot leave patients without proper cauterization - incised limbs will bleed about 5x as much as a limb with internal bleeding.

### Surgeries
**Internal bleeding**
*Fixed by doing the vascular repair surgery on the affected limb.*
Steps:
1. Scalpel
2. Retractor
3. Hemostat
4. Fix o' vein
5. Cautery

**Fractures of any kind**
*Fixed by doing the fracture repair surgery on the affected limb.*
Steps:
1. Scalpel
2. Retractor
3. Hemostat
4. Bone setter
5. Bone gel
6. Cautery

**Limb augmentation**
*You may have noticed that well, you can't print robotic hands or feet. Don't worry, i did a very crappy solution. If you want to augment a hand, just do the surgery as normal - when it asks for the augmentation step, just use the correct arm or leg on the patient's hand or foot. That will replace only the chosen limb. Augmenting an arm will however, augment both the arm and hand, same logic applies for legs*

**Limb reattachment**
*Some of the logic described above applies here. Simply put, reattaching an arm will also reattach the hand, if the arm has a hand attached to it that is. Most of the time you won't need two surgeries for limb reattachment.*

### Final notes
**I am completely aware that testing so far has proven this to be very buggy and unstable. It might suck at the moment, but i do ask patience on your part, reader - I am fixing reported problems as soon as i can, and i am indeed taking suggestions into account. This might be a very bumpy, terrible ride - with accidents in the middle of the way - but it is for good reason. TG medicine is too simple and not suited for an RP server, it carries little consequence and leaves medbay as an useless department. This terrible piece of code might actually seem even worse than that right now, as it barely functions, but it will be eventually fixed and playable.**

### CODE REVIEW

**Comment by ShadeAware** [comment]
Poggers

**Comment by SquishyOranges** [comment]
👍

File: code/game/objects/items/stacks/medical.dm [hunk 1]
```diff
@@ -165,3 +166,91 @@
...
+	icon_state = "tribal_splint"
+	desc = "Ooga booga rock crush bone."
+	self_delay = 200
+
+/obj/item/stack/medical/nanopaste
+	name = "nanite paste"
+	singular_name = "nanite paste"
+	icon = 'modular_skyrat/icons/obj/medical.dmi'
+	icon_state = "nanopaste"
+	desc = "01110111 01100101 01100101 01100100 00100000 01100101 01100001 01110100 01100101 01110010."
```
Comment (ShadeAware) [inline code/game/objects/items/stacks/medical.dm:228]: "weed eater" isn't exactly a good description of something you use to repair artificial limbs and organs.

**Comment by Owai-Seek** [comment]
Undocumented change to a traitor bundle, reducing it's cost and adding another item to it.

We already have Synthmeat, and Meat Product, I don't feel like we need to add another way to make meat.

I don't like the idea of having to put two limbs on instead of one, but that's just like, my opinion. (Arm/Hand. Foot/Leg) but I'd have to see it in action before I really have a final judgement on it.

Otherwise, my only major concern is there's no way for people to learn how to do medical besides a lot of trial and error, and when it comes to how long it takes people to get back to what they were doing before they got exploded/carp'd, it ends up with them being incredibly frustrated and mad at medical.

File: code/game/objects/items/stacks/medical.dm (hunk 1 above)
Comment (BobJoga) [reply code/game/objects/items/stacks/medical.dm:228]: I thought it was a mildly funny veggietales reference but yeah, i'm gonna change it.

**Comment by BobJoga** [comment]
> Undocumented change to a traitor bundle, reducing it's cost and adding another item to it.
> 
> We already have Synthmeat, and Meat Product, I don't feel like we need to add another way to make meat.
> 
> I don't like the idea of having to put two limbs on instead of one, but that's just like, my opinion. (Arm/Hand. Foot/Leg) but I'd have to see it in action before I really have a final judgement on it.
> 
> Otherwise, my only major concern is there's no way for people to learn how to do medical besides a lot of trial and error, and when it comes to how long it takes people to get back to what they were doing before they got exploded/carp'd, it ends up with them being incredibly frustrated and mad at medical.

To address your concerns:
First off - Yeah i should've just modified the bundle too make sure it works, not anything balance related. I will fix this in a jiffy.
Secondly - I added biomeat especifically for the purpose of creating biomass - other types of meat are not nearly as effective. The initial idea was for only biomeat to be used for biomass, but people complained about it in discord so i just made biomeat the most effective, but you can still use other meats.
Thridly - If i understand you correctly, don't worry about that. You can indeed just attach and arm, if it has a hand attached it will simply put install both the arm and hands. In most normal circunstances, you're really just gonna have to do a single reattachment surgery - unless the hand got lost somehow.

**Comment by CinderWC** [comment]
holy fuck the madlad did it

**Comment by EldritchSigma** [comment]
Boneman good. Though I am kind of concerned how this will play into balance with lavaland (bubblegum blood grasp comes to mind).

**Comment by BobJoga** [comment]
> Boneman good. Though I am kind of concerned how this will play into balance with lavaland (bubblegum blood grasp comes to mind). Maybe have survival medipens have a chem that temporary fix your bones or another medipen that just does it so you dont have to jaunt out if you get clipped (its sometimes simply inevitable)

legion cores temporarily remove slowdown because of the citadel nerf. i'll see if any major problems are caused by this in lavaland with the testmerge, and fix them.

**Comment by Addirem** [comment]
Cum man, cum me some cum

**Comment by Investigator77** [comment]
Chad

**Comment by BobJoga** [comment]
Closing this for now since i'll make a more updated and better version and remake the description and things on the PR. The dream hasn't died.

**Pull request closed without merging**